source/common/network/connection_impl.cc

```diff
@@ -395,9 +396,35 @@ void ConnectionImpl::updateWriteBufferStats(uint64_t num_written, uint64_t new_s
                                            buffer_stats_->write_current_);
 }
 
+const std::string Network::ConnectionImpl::destinationAddress() {
```

Comment:
per the previous comment on just having destinationAddress(), can we form the address in the constructor, and then just store it as a string like tcp://1.2.3.4:80 blah. Basically, this code can't assume that a connection is a TCP connection. It might not be.